java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -538,10 +539,12 @@ abstract class AbstractArcHost(
      *
      * @property identifier a [ParticleIdentifier] from a [Plan.Particle] spec
      */
-    open suspend fun instantiateParticle(identifier: ParticleIdentifier): Particle {
-        return particleConstructors[identifier]?.invoke() ?: throw IllegalArgumentException(
-            "Particle $identifier not found."
-        )
+    open suspend fun instantiateParticle(
+        identifier: ParticleIdentifier,
+        spec: Plan.Particle? = null
```

Comment:
why default null, this looks dangerous.

Reply:
Null will still need to be passed by callers however. There's no way around it until we can serialize schemas :(

Reply:
I know, but should it be a default? I'm worried that it will allow calling instantiateParticle(identifier) without realising that there should be a second argument.

Reply:
Yes, I agree, there are only 2 callsites, so you can refactor both to pass the null explicitly.